.github/workflows/go.yml

```diff
@@ -1,5 +1,11 @@
 name: Go
-on: [push]
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
it is copied from the starter template.

https://github.com/actions/starter-workflows/blob/0bba821a5dfe745cb0c243d0d9bfe8218005dc1b/ci/go.yml#L3-L7